codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2StreamFrame.java

```diff
@@ -15,24 +15,29 @@
  */
 package io.netty.handler.codec.http2;
 
+import io.netty.util.internal.ObjectUtil;
 import io.netty.util.internal.UnstableApi;
 
 /**
  * Abstract implementation of {@link Http2StreamFrame}.
  */
 @UnstableApi
 public abstract class AbstractHttp2StreamFrame implements Http2StreamFrame {
-    private Object stream;
+
+    private int streamId = -1;
 
     @Override
-    public AbstractHttp2StreamFrame setStream(Object stream) {
-        this.stream = stream;
+    public AbstractHttp2StreamFrame setStreamId(int streamId) {
+        if (this.streamId != -1) {
+            throw new IllegalStateException("Stream identifier may only be set once.");
+        }
+        this.streamId = ObjectUtil.checkPositiveOrZero(streamId, "Stream id must be non-negative.");
```

Comment:
@buchgr actually the second arg is the name of the field and not an exception message ;)